x-pack/legacy/plugins/siem/public/pages/hosts/details/details_tabs.test.tsx

```diff
@@ -33,39 +35,28 @@ jest.mock('../../../containers/source', () => ({
 }));
 
 describe('body', () => {
-  test('render snapshot', () => {
-    const child: CommonChildren = () => <span>{'I am a child'}</span>;
-    const wrapper = shallow(
-      <TestProviders>
-        <HostDetailsBody
-          children={child}
-          from={0}
-          isInitializing={false}
-          detailName={'host-1'}
-          setQuery={() => {}}
-          to={0}
-        />
-      </TestProviders>
-    );
-    expect(toJson(wrapper)).toMatchSnapshot();
-  });
-
   test('it should pass expected object properties to children', () => {
-    const child = jest.fn();
-    mount(
+    const wrapper = mount(
       <TestProviders>
-        <HostDetailsBody
-          children={child}
-          from={0}
-          isInitializing={false}
-          detailName={'host-1'}
-          setQuery={() => {}}
-          to={0}
-        />
+        <MemoryRouter initialEntries={[`/hosts/host-1/authentications`]}>
+          <HostDetailsTabs
+            from={0}
+            isInitializing={false}
+            detailName={'host-1'}
+            setQuery={() => {}}
+            to={0}
+            setAbsoluteRangeDatePicker={(jest.fn() as unknown) as SetAbsoluteRangeDatePicker}
+            hostDetailsPagePath={hostDetailsPagePath}
+            indexPattern={mockIndexPattern}
+            type={type}
+            filterQuery='{"bool":{"must":[],"filter":[{"match_all":{}},{"match_phrase":{"host.name":{"query":"host-1"}}}],"should":[],"must_not":[]}}'
+          />
+        </MemoryRouter>
       </TestProviders>
     );
+
     // match against everything but the functions to ensure they are there as expected
-    expect(child.mock.calls[0][0]).toMatchObject({
+    expect(wrapper.find('AuthenticationsQueryTabBody').props()).toMatchObject({
```

Comment:
Given that we've made the child components specific here, what about let's make each of them a test case, so that we can also guarantee the existing routes.

Reply:
Added all cases, Thanks!